packages/nameguard-react/src/components/Report/NoGraphemesWarning.tsx

```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import cc from "classcat";
-import { CheckIcon } from "./CheckIcon";
+import { ResultCodeIcon } from "./ResultCodeIcon";
```

Comment:
`ResultCodeIcon` is too generic. It sounds like it could be an icon for any kind of result code. But this icon is specifically for the purpose of visualizing the result from a NameGuard check. Ex: a `CheckResultCode`.

I'm ok to rename this component to `CheckResultIcon` or `CheckResultCodeIcon`.

Let's take a step back. What is NameGuard doing? It's important to understand otherwise all these names are just meaningless noise.

NameGuard performs a series of "checks" on names, labels, and graphemes. Each of these "checks" produces a result in the form of a `CheckResultCode`. Each `CheckResultCode` can be visualized in a UI using a `CheckResultIcon`.

NameGuard also provides "ratings" on names, labels, and graphemes. The `Rating` assigned to a name, label, or grapheme is a function of all the results of all the checks performed on that name, label, or grapheme.

A check result is fundamentally different from a rating. Let's please completely avoid smashing these ideas together.

Reply:
@lightwalker-eth we then need an idea to create `CheckResultIcon`s for objects that provide a `null` `highest_risk`, which is exactly the case described in one of your loom videos where `Looks good` didn't have an icon representation. Before this PR was created, a plain string was used to inform "pass" or "info", for example, for the `CheckResultIcon` code and this is why it used to work "without problems". When `CheckResultCode` was introduced as `code` type the things broke because `CheckResultCode`  is only provided when `highest_risk` is defined, otherwise, we only have a `Rating` (from `ConsolidatedNameGuardReport.rating` to trust on). I don't see how we can give support to all use cases without:

1. Making `CheckResultIcon` receive either a `CheckResultCode` or a `Rating` OR
2. Creating two dedicated components, one for `CheckResultCode` and one for `Rating`, to render the icons that currently `CheckResultIcon` already renders